__tests__/api/BundleContent/Bundle.test.ts

```diff
@@ -130,30 +135,693 @@ describe("Bundle01", () => {
+
+        // Mocks for the manifest - META-INF exists
+        const existsSpy = jest.spyOn(fs, "existsSync").mockImplementationOnce(() => ( false ));
+        // Mocks for the manifest - Bundle dir writable
+        const accessSpy = jest.spyOn(fs, "accessSync").mockImplementationOnce(() => ( true ));
+
+        let err: Error;
+        try {
+          const bund = new Bundle("__tests__/__resources__/ExampleBundle01", false, false);
+          bund.prepareForSave();
+        }
+        catch (error) {
+          err = error;
+        }
+
+        expect(err).toBeUndefined();
+        expect(existsSpy).toHaveBeenCalledTimes(1);
+        expect(accessSpy).toHaveBeenCalledTimes(1);
+    });
+    it("should complain if no write permission to bundle directory", () => {
+
+        // Mocks for the manifest - META-INF exists
+        const existsSpy = jest.spyOn(fs, "existsSync").mockImplementationOnce(() => ( false ));
+        // Mocks for the manifest - Bundle dir writable
+        const accessSpy = jest.spyOn(fs, "accessSync").mockImplementationOnce(() => { throw new Error("Wibble"); });
+
+        let err: Error;
+        try {
+          const bund = new Bundle("__tests__/__resources__/ExampleBundle01", false, false);
+          bund.prepareForSave();
+        }
+        catch (error) {
+          err = error;
+        }
+
+        expect(existsSpy).toHaveBeenCalledTimes(1);
+        expect(accessSpy).toHaveBeenCalledTimes(1);
+        expect(err).toBeDefined();
+        expect(err.message).toContain("cics-deploy requires write permission to: ");
+    });
+    it("should complain if no write permission to META-INF directory", () => {
+
+        // Mocks for the manifest - META-INF exists
+        const existsSpy = jest.spyOn(fs, "existsSync").mockImplementationOnce(() => ( true ));
+        // Mocks for the manifest - META-INF is writable
+        const accessSpy = jest.spyOn(fs, "accessSync").mockImplementationOnce(() => { throw new Error("Wibble"); });
+
+        let err: Error;
+        try {
+          const bund = new Bundle("__tests__/__resources__/ExampleBundle01", false, false);
+          bund.prepareForSave();
+        }
+        catch (error) {
+          err = error;
+        }
+
+        expect(existsSpy).toHaveBeenCalledTimes(1);
+        expect(accessSpy).toHaveBeenCalledTimes(1);
+        expect(err).toBeDefined();
+        expect(err.message).toContain("cics-deploy requires write permission to: ");
+    });
+    it("should complain if no overwrite permission to manifest", () => {
+
+        // Mocks for the manifest - META-INF exists
+        const existsSpy = jest.spyOn(fs, "existsSync").mockImplementationOnce(() => ( true ));
+        // Mocks for the manifest - META-INF is writable
+        const accessSpy = jest.spyOn(fs, "accessSync").mockImplementationOnce(() => ( true ));
+        // Mocks for the manifest - manifest exists
+        jest.spyOn(fs, "existsSync").mockImplementationOnce(() => ( true ));
```

Comment:
As described in other comment, suggest converting this to a single call to `mockImplementationOnce()` for `existsSync` that checks the path.